### PR TITLE
logging: Removing logging v1 support from all backends

### PR DESCRIPTION
**Pull request by nordic-krch**
With this PR a process of removal of logging v1 is starting. Here support for v1 is removed from all backends.

### CODE REVIEW

File: subsys/bluetooth/host/monitor.c [hunk 1]
```diff
@@ -310,47 +310,6 @@ static inline uint8_t monitor_priority_get(uint8_t log_level)
 	return BT_LOG_DBG;
 }
 
-static void monitor_log_put(const struct log_backend *const backend,
```
Comment (jori-nordic) [inline subsys/bluetooth/host/monitor.c:313]: I think you meant to delete `monitor_log_process` instead.

Comment (nordic-krch) [reply subsys/bluetooth/host/monitor.c:313]: no, `put` is a legacy v1 backend API and process is the one working with v2.

Comment (jori-nordic) [reply subsys/bluetooth/host/monitor.c:313]: but at L360 (new version), you're referencing a function that you are deleting ?

File: subsys/bluetooth/host/monitor.c [hunk 2]
```diff
@@ -398,8 +357,7 @@ static void monitor_log_init(const struct log_backend *const backend)
 }
 
 static const struct log_backend_api monitor_log_api = {
-	.process = IS_ENABLED(CONFIG_LOG2_DEFERRED) ? monitor_log_process : NULL,
-	.put = IS_ENABLED(CONFIG_LOG1_DEFERRED) ? monitor_log_put : NULL,
+	.put = monitor_log_put,
```
Comment (dcpleung) [inline subsys/bluetooth/host/monitor.c:360]: Removed the wrong line?

Comment (nordic-krch) [reply subsys/bluetooth/host/monitor.c]: fixed.

File: subsys/bluetooth/host/monitor.c (hunk 1 above)
Comment (nordic-krch) [reply subsys/bluetooth/host/monitor.c:313]: my bad. Fixed.

**Review by jori-nordic — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**